src/http/cookie.cr

```diff
@@ -217,83 +300,161 @@ module HTTP
     # Gets the current `HTTP::Cookie` for the given *key*.
     #
     # ```
-    # request.cookies["foo"].value # => "bar"
+    # require "http/cookie"
+    #
+    # cookies = HTTP::Cookies{HTTP::Cookie.new("session", "god")}
+    # cookies["session"] # => <HTTP::Cookie @name="session", @value="god" ... >
+    # cookies["debug"]   # => raises KeyError
     # ```
-    def [](key)
+    def [](key : String) : Cookie
       @cookies[key]
     end
 
     # Gets the current `HTTP::Cookie` for the given *key* or `nil` if none is set.
     #
     # ```
-    # require "http/client"
+    # require "http/cookie"
     #
-    # request = HTTP::Request.new "GET", "/"
-    # request.cookies["foo"]? # => nil
-    # request.cookies["foo"] = "bar"
-    # request.cookies["foo"]?.try &.value # > "bar"
+    # cookies = HTTP::Cookies.new{HTTP::Cookie.new("session", "god")}
+    # cookies["session"]? # => <HTTP::Cookie @name="session", @value="god" ... >
+    # cookies["debug"]?   # => false
     # ```
-    def []?(key)
+    def []?(key : String) : Cookie?
       @cookies[key]?
     end
 
     # Returns `true` if a cookie with the given *key* exists.
     #
     # ```
-    # request.cookies.has_key?("foo") # => true
+    # require "http/cookie"
+    #
+    # cookies = HTTP::Cookies.new{HTTP::Cookie.new("session", "god")}
+    # cookies.has_key?("session") # => true
+    # cookies.has_key?("user")    # => false
     # ```
-    def has_key?(key)
+    def has_key?(key : String) : Bool
       @cookies.has_key?(key)
     end
 
-    # Adds the given *cookie* to this collection, overrides an existing cookie
+    # Adds the given *cookie* to this collection indexed as the name of the cookie.  If another cookie is added with the same name it will overwrite the older cookie.
```

Comment:
```suggestion
    # Adds the given *cookie* to this collection indexed as the name of the cookie.  If another cookie with the same name already exists, it is overriden.
```
IMO it's better to describe what this method does *now* not a future interaction.